core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/clone/WorkspaceCloningRunner.scala

```diff
@@ -0,0 +1,160 @@
+      SOURCE_WORKSPACE_KEY -> sourceWorkspace.workspaceIdAsUUID.toString,
+      DISABLE_AUTOMATIC_APP_CREATION_KEY -> disableAutoAppCreate
+    )
+
+    request.copyFilesWithPrefix match {
+      case Some(prefix) => args + (STORAGE_CONTAINER_CLONE_PREFIX_KEY -> prefix)
+      case None         => args
+    }
+  }
+
+  def getInitialWSMJobId(args: Option[Map[String, String]]): Option[String] =
+    args.flatMap(argsMap => argsMap.get(WORKSPACE_INITIAL_CLONE_JOBID_KEY))
+
+  def isAutomaticAppCreationDisabled(args: Option[Map[String, String]]): Boolean =
+    args.flatMap(argsMap => argsMap.get(DISABLE_AUTOMATIC_APP_CREATION_KEY)).getOrElse("false") == "true"
+
+  def getSourceWorkspaceId(args: Option[Map[String, String]]): Option[UUID] =
+    args.flatMap(argsMap => argsMap.get(SOURCE_WORKSPACE_KEY)).map(id => UUID.fromString(id))
+
+  def getStorageContainerClonePrefix(args: Option[Map[String, String]]): Option[String] =
+    args.flatMap(argsMap => argsMap.get(STORAGE_CONTAINER_CLONE_PREFIX_KEY))
+}
+
+class WorkspaceCloningRunner(
+  val samDAO: SamDAO,
+  val gcsDAO: GoogleServicesDAO,
+  leonardoDAO: LeonardoDAO,
+  workspaceManagerDAO: WorkspaceManagerDAO,
+  monitorRecordDao: WorkspaceManagerResourceMonitorRecordDao,
+  workspaceRepository: WorkspaceRepository
+) extends WorkspaceManagerResourceJobRunner
+    with LazyLogging
+    with UserCtxCreator {
+
+  def cloneFail(wsId: UUID, message: String)(implicit executionContext: ExecutionContext): Future[Int] =
+    workspaceRepository.setFailedState(wsId, WorkspaceState.CloningFailed, message)
+
+  override def apply(job: WorkspaceManagerResourceMonitorRecord)(implicit
+    executionContext: ExecutionContext
+  ): Future[WorkspaceManagerResourceMonitorRecord.JobStatus] = {
+    def logFailure(msg: String, t: Option[Throwable] = None): Unit = {
+      val logMessage = s"CloneWorkspace monitoring job with id ${job.jobControlId} failed: $msg"
+      t match {
+        case Some(t) => logger.error(logMessage, t)
+        case None    => logger.error(logMessage)
+      }
+    }
+
+    val workspaceId = job.workspaceId match {
+      case Some(id) => id
+      case None =>
+        logger.error(
+          s"Job to monitor workspace deletion created with id ${job.jobControlId} but no workspace ID"
+        )
+        return Future.successful(Complete)
+    }
+
+    val userEmail = job.userEmail match {
+      case Some(email) => email
+      case None =>
+        val msg =
+          s"Unable to update clone status for workspace $workspaceId because no user email set on monitoring job"
+        logFailure(msg)
+        return cloneFail(workspaceId, msg).map(_ => Complete)
+    }
+
+    getUserCtx(userEmail).transformWith {
+      case Failure(t) =>
+        val msg =
+          s"Unable to retrieve clone workspace results for workspace $workspaceId: unable to retrieve request context for $userEmail"
+        logFailure(msg, Some(t))
+        cloneFail(workspaceId, msg).map(_ => Incomplete)
```

Comment:
Yes, this leaves the job in a retryable state, but the error shows up on the workspace. 

It's the same pattern we use for this type of failure in all our other runners. 
The rational being that these are probably transient errors, like permission propagation delays, or connection issues, but if it happens, we want to see it on the workspace, in case it isn't transient.